src/services/search/dealerSearch.ts

```diff
@@ -0,0 +1,40 @@
+import { postData, Service } from "../../base"
+
+import { Paginated } from "../../types/pagination"
+import { DealerParams } from "../../types/params/dealer"
+
+import { SearchDealer } from "../../types/models/dealer"
+
+const defaultPagination = {
+  page: 0,
+  size: 3,
+}
+
+const searchForDealers = (path, searchQuery: DealerParams) => {
+  const { pagination, sort, query } = searchQuery
+  const { page, size } = pagination
+
+  const sizeOrDefault =
+    parseInt((size || "").toString(), 10) || defaultPagination.size
+  const pageOrDefault =
+    parseInt((page || "").toString(), 10) - 1 || defaultPagination.page
+
+  const body = {
+    pagination: {
+      page: pageOrDefault,
+      size: sizeOrDefault,
+    },
+    sort: [sort],
+    query,
+  }
+
+  return postData(Service.SEARCH, path, body)
```

Comment:
What happens if the post request fails? Like do we need to use the error handler as for other api calls?

Reply:
👍  changed it

Reply:
but my tests are failling so maybe i need some help there. 😢

Reply:
https://github.com/carforyou/carforyou-api-client-pkg/blob/master/src/services/search/listingSearch.ts#L78
here and in many other places we don't take care of that

Reply:
I have strong feelings here: the handling of errors is responsibility of the project, not the API client :)

Reply:
Ahh I see messed that up with the validation errors that are somtims enveloped by the api client